tests/integration/questionnaire/test_questionnaire_list_collector.py

```diff
@@ -320,3 +320,170 @@ def test_adding_person_using_second_list_collector_when_no_people(
         self.get(first_person_remove_link)
 
         self.assertInBody("Are you sure you want to remove this person?")
+
+    def test_adding_from_the_summary_page_adds_the_return_to_param_to_the_url(
+        self,
+    ):
+        self.launchSurvey("test_list_collector")
+
+        self.assertInBody("Does anyone else live here?")
+
+        self.post({"anyone-else": "Yes"})
+
+        self.add_person("Marie Claire", "Doe")
+
+        self.assertInSelector("Marie Claire Doe", "[data-qa='list-item-1-label']")
+
+        self.post({"anyone-else": "No"})
+
+        self.post()
+
+        self.post({"another-anyone-else": "No"})
+
+        self.assertInBody("List Collector Summary")
+
+        # Make another mistake
+
+        add_link = self.get_add_someone_link()
+
+        self.get(add_link)
+
+        self.assertInUrl("?return_to=section-summary")
+
+    def test_removing_from_the_summary_page_adds_the_return_to_param_to_the_url(
+        self,
+    ):
+        self.launchSurvey("test_list_collector")
+
+        self.assertInBody("Does anyone else live here?")
+
+        self.post({"anyone-else": "Yes"})
+
+        self.add_person("Marie Claire", "Doe")
+
+        self.assertInSelector("Marie Claire Doe", "[data-qa='list-item-1-label']")
+
+        self.post({"anyone-else": "No"})
+
+        self.post()
+
+        self.post({"another-anyone-else": "No"})
+
+        self.assertInBody("List Collector Summary")
+
+        # Make another mistake
+
+        remove_link = self.get_link("remove", 1)
+
+        self.get(remove_link)
+
+        self.assertInUrl("?return_to=section-summary")
+
+    def test_changing_item_from_the_summary_page_adds_the_return_to_param_to_the_url(
+        self,
+    ):
+        self.launchSurvey("test_list_collector")
+
+        self.assertInBody("Does anyone else live here?")
+
+        self.post({"anyone-else": "Yes"})
+
+        self.add_person("Marie Claire", "Doe")
+
+        self.assertInSelector("Marie Claire Doe", "[data-qa='list-item-1-label']")
+
+        self.post({"anyone-else": "No"})
+
+        self.post()
+
+        self.post({"another-anyone-else": "No"})
+
+        self.assertInBody("List Collector Summary")
+
+        change_link = self.get_link("change", 1)
+
+        self.get(change_link)
+
+        self.assertInUrl("?return_to=section-summary")
+
+    def test_adding_from_the_summary_page_and_then_removing_from_parent_page_keeps_return_to_url_param(
```

Comment:
@MebinAbraham This test and the one below ensure the URL param remains when changing and removing from the list collector, after adding from the summary.